firmware/src/core/utils/Utils.cpp

```diff
@@ -291,4 +294,47 @@ void Utils::colorizeImageData(uint16_t *pixels565, size_t length, uint32_t targe
         // Map grayscale to the target color
         pixels565[i] = grayscaleToTargetColor(grayscale, targetR8, targetG8, targetB8, brightness, swapBytes);
     }
+}
+
+char *Utils::copyString(const std::string &originalString) {
```

Comment:
This function is a memory leak waiting to happen.
Why would strings need to be copied like this?

Reply:
The WiFiManager does not copy the parameters into its own local buffers, ie. if the variable that we use as parameter goes out of scope, we have a problem. 
This is just a convenient method to create a buffer, put the contents of the string in it and return the pointer to it.
But I'll rename the function to make it more clear.